python/MaterialX/__init__.py

```diff
@@ -1,3 +1,24 @@
+# Python 3.8+ on Windows: DLL search paths for dependent
+# shared libraries
+# Refs.:
+# - https://github.com/python/cpython/issues/80266
+# - https://docs.python.org/3.8/library/os.html#os.add_dll_directory
+import os
+import sys
+if sys.platform == "win32" and sys.version_info >= (3, 8):
+    import importlib.metadata
+    try:
+        importlib.metadata.version('MaterialX')
+    except importlib.metadata.PackageNotFoundError:
+        # On a non-pip installation, this file is in %INSTALLDIR%\python\MaterialX
+        # We need to add %INSTALLDIR%\bin to the DLL path.
+        mxdir = os.path.dirname(__file__)
+        pydir = os.path.split(mxdir)[0]
+        installdir = os.path.split(pydir)[0]
```

Comment:
Assuming `os.path.split` performs no split operation when the input path ends in a trailing slash, does this second invocation of `os.path.split` actually have an effect on the result?

Reply:
On a non-pip install, the file is `"%INSTALLDIR%\python\MaterialX\_init_.py"`.
os.path.dirname() never ends with a trailing slash, so it will return `"%INSTALLDIR%\python\MaterialX"`.
First os.path.split() will return `("%INSTALLDIR%\python", "MaterialX")`.
Second os.path.split() will return `("%INSTALLDIR%", "python")`.
From there we can find the bin dir and add it to the DLL load search path.